src/test/java/com/jcabi/s3/AwsBucketTest.java

```diff
@@ -55,4 +59,22 @@ public void findsAndReturnsOckets() throws Exception {
         MatcherAssert.assertThat(ocket, Matchers.notNullValue());
     }
 
+    /**
+     * AwsBucket can check if the bucket exists.
+     * @throws IOException If fails
+     */
+    @Test(expected = IOException.class)
+    public void checkBucketExistence() throws IOException {
+        final String bucketName = "example1.com";
+        final Region region = Mockito.mock(Region.class);
+        final AmazonS3 aws = Mockito.mock(AmazonS3.class);
+        Mockito.when(region.aws()).thenReturn(aws);
+        Mockito.when(aws.doesBucketExist(bucketName)).thenReturn(true, false)
+            .thenThrow(new AmazonServiceException("Test exception"));
+        final Bucket bucket = new AwsBucket(region, bucketName);
+        Assert.assertTrue(bucket.exists());
+        Assert.assertFalse(bucket.exists());
+        bucket.exists();
```

Comment:
you're testing three different scenarios in one test method. this is in general an anti-pattern. if something will fail, it will be difficult to understand what exactly is wrong and how exactly the class is broken. I would suggest to break this test method into three smaller ones.